src/order/ideal.lean

```diff
@@ -0,0 +1,136 @@
+/-
+Copyright (c) 2020 David Wärn. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: David Wärn
+-/
+import order.basic
+import data.equiv.encodable
+
+/-!
+# Order ideals, cofinal sets, and Rasiowa–sikorski
+
+## Main definitions
+
+We work with a preorder `P` throughout.
+
+- `ideal P`: the type of upward directed, downward closed subsets of `P`.
+             Dual to the notion of a filter on a preorder.
+- `cofinal P`: the type of subsets of `P` containing arbitrarily large elements.
+               Could also be called 'upward dense'.
+- `ideal_of_countable_cofinals p 𝒟`, where `p : P`, and `𝒟` is a countable family
+  of cofinal subsets of P: an ideal in `P` which contains `p` and intersects every
+  set in `𝒟`.
+
+## References
+
+https://en.wikipedia.org/wiki/Rasiowa–Sikorski_lemma
+
+Note that Wikipedia uses the opposite ordering on `P`.
+
```

Comment:
Is there also a reference on order ideals you used?

Reply:
Added!